### ISSUE TITLE
max_size是自己定的吗

### ISSUE THREAD

**Issue by wushupei1997** [BADBADBADBOY/pytorch.ctpn#9]
代码中max_size的值可以根据自己的数据集定吗，还是一定要按照源代码中的600，1200

**Comment by fxwfzsxyq** [comment]
> 代码中max_size的值可以根据自己的数据集定吗，还是一定要按照源代码中的600，1200

可以，但是一般和你训练时的size差距不要太大，一般等于或稍大于，比如我训练max_size 1048,测试设置了1200

**Comment by wushupei1997** [comment]
![image](https://user-images.githubusercontent.com/57393009/83480564-58e76780-a4ce-11ea-83a3-a8a5ac0c193e.png)
感谢！还有一个问题是我在复现代码时发现的，代码出现IndexError: Out of bounds on buffer access (axis 1)这种问题，请问这是为什么呢？对源代码我没有改动其他地方

**Comment by fxwfzsxyq** [comment]
> ![image](https://user-images.githubusercontent.com/57393009/83480564-58e76780-a4ce-11ea-83a3-a8a5ac0c193e.png)
> 感谢！还有一个问题是我在复现代码时发现的，代码出现IndexError: Out of bounds on buffer access (axis 1)这种问题，请问这是为什么呢？对源代码我没有改动其他地方

错误位置在哪一句代码，我觉得是你的标注框有问题

**Comment by fxwfzsxyq** [comment]
> ![image](https://user-images.githubusercontent.com/57393009/83480564-58e76780-a4ce-11ea-83a3-a8a5ac0c193e.png)
> 感谢！还有一个问题是我在复现代码时发现的，代码出现IndexError: Out of bounds on buffer access (axis 1)这种问题，请问这是为什么呢？对源代码我没有改动其他地方

应该是你的标注框有问题，你加个try catch ，报错的时候输出下你的gt_boxes

**Comment by wushupei1997** [comment]
![image](https://user-images.githubusercontent.com/57393009/83492285-a1f6e600-a4e5-11ea-80ba-0fa0f3c8a639.png)
这个标注框是有什么问题吗？≧ ﹏ ≦

**Comment by fxwfzsxyq** [comment]
> ![image](https://user-images.githubusercontent.com/57393009/83492285-a1f6e600-a4e5-11ea-80ba-0fa0f3c8a639.png)
> 这个标注框是有什么问题吗？≧ ﹏ ≦

为什么你的这个是三维的，应该有四维，你是不是改动过什么了，正确的应该是这样的[[1,2,3,4],[1,2,3,4]]，你自己调试看看为什么会变成三维的

**Comment by wushupei1997** [comment]
![image](https://user-images.githubusercontent.com/57393009/83495762-b7bada00-a4ea-11ea-9649-d07aaf8008d6.png)

查了一下代码，是在utils\rpn_msr\anchor_target_layer.py，gt_boxes = gt_boxes[:,:-1]这句代码使得四维变成了三维，不加这句代码的话，是四维的；但是把这句注掉，
![image](https://user-images.githubusercontent.com/57393009/83496014-09fbfb00-a4eb-11ea-8836-4e69fa851cda.png)
assert gt_rois.shape[1] == 5这句又会报错AssertionError，就很迷
![image](https://user-images.githubusercontent.com/57393009/83496101-2a2bba00-a4eb-11ea-915d-bd719908cc8d.png)

**Comment by wushupei1997** [comment]
问一下答主，你的gt_boxes 是不是五维的，要不然gt_boxes = gt_boxes [：，：-1]这句话应该会使维数降低吧

**Comment by fxwfzsxyq** [comment]
> ![image](https://user-images.githubusercontent.com/57393009/83495762-b7bada00-a4ea-11ea-9649-d07aaf8008d6.png)
> 
> 查了一下代码，是在utils\rpn_msr\anchor_target_layer.py，gt_boxes = gt_boxes[:,:-1]这句代码使得四维变成了三维，不加这句代码的话，是四维的；但是把这句注掉，
> ![image](https://user-images.githubusercontent.com/57393009/83496014-09fbfb00-a4eb-11ea-8836-4e69fa851cda.png)
> assert gt_rois.shape[1] == 5这句又会报错AssertionError，就很迷
> ![image](https://user-images.githubusercontent.com/57393009/83496101-2a2bba00-a4eb-11ea-915d-bd719908cc8d.png)

我知道你的问题是什么了，
你的标注是我这种格式吗，x1,y1,x2,y2,x3,y3,x4,y4,label，这里的label必须要有，我这里加入了模糊字符忽略不参与loss计算，所以我生成的的gt的auchor是多一维的，gt_boxes应该传进来是个6维的，你去https://github.com/BADBADBADBOY/pytorch.ctpn/blob/master/dataLoader/dataLoad.py 77行看看

**Comment by fxwfzsxyq** [comment]
> 问一下答主，你的gt_boxes 是不是五维的，要不然gt_boxes = gt_boxes [：，：-1]这句话应该会使维数降低吧

上面的说明忘记改了应该是6维的，多了一个忽略标志，我是判断label中含有#，就忽略这个标注文本，类似icdar2015，所以需要训练的文本的label只要不包含#，随意什么都可以，或者根据自己需求自行修改dataload

**Comment by wushupei1997** [comment]
奥，原来如此，这样的话，这个label是不是就是文本的内容，那如果我的label都不包含#，是不是也可以训练？

**Comment by fxwfzsxyq** [comment]
> 奥，原来如此，这样的话，这个label是不是就是文本的内容，那如果我的label都不包含#，是不是也可以训练？

是的，只要你是这种格式的数据，应该都没问题，只是包含#的文本不参与文本检测的训练，不包含的参与训练

**Comment by wushupei1997** [comment]
感谢感谢！
